src/swap/SwapExecuteScreen.tsx

```diff
@@ -118,33 +152,23 @@ export function SwapExecuteScreen() {
       case SwapState.START:
       case SwapState.APPROVE:
       case SwapState.EXECUTE:
-        return (
-          <ActivityIndicator
-            size="large"
-            color={colors.greenBrand}
-            testID="SwapExecuteScreen/loadingIcon"
-            style={styles.activityIndicator}
-          />
-        )
+        return <GreenLoadingSpinner />
       case SwapState.ERROR:
       case SwapState.PRICE_CHANGE:
-        return (
-          <View
-            testID="SwapExecuteScreen/errorIcon"
-            style={[styles.iconContainer, { backgroundColor: colors.warning }]}
-          >
-            <Times color={colors.light} />
-          </View>
-        )
+        return <RedLoadingSpinnerToInfo />
       case SwapState.COMPLETE:
-        return (
-          <View
-            testID="SwapExecuteScreen/completeIcon"
-            style={[styles.iconContainer, { backgroundColor: colors.greenUI }]}
-          >
-            <Checkmark color={colors.light} />
-          </View>
-        )
+        return <GreenLoadingSpinnerToCheck />
+    }
+  }, [swapState])
+
+  const NavigationButtons = useMemo(() => {
+    switch (swapState) {
+      case SwapState.ERROR:
+        return <TryAgainAndDone />
+      case SwapState.COMPLETE:
+        return <DoneAndSwapAgain />
+      default:
+        return <View style={[styles.actionBar, { minHeight: 130 }]} />
```

Comment:
is this 130 magic number to ensure that the animated loader stays in the same place when there are no buttons? this seems a little brittle (e.g. if we ever updated the size of the buttons, this seems like an easy place to break) - but i also don't think it works for the error scenario since you've got `contentContainer` to position content in the middle, the error content is longer than the pending content so the icon does move around.

i think to keep the icon in the same place, you need to have `contentContainer` position content with `flex-start` so that you can always control where the top of the icon is. this might mean that the content is not vertically positioned on the screen, but i think that is a worthwhile tradeoff. what do you think?